### ISSUE TITLE
Failed to parse (WS-2902) - index out of range [0] with length 0

### ISSUE THREAD

**Issue by iBotPeaches** [tedpearson/ambientweatherexporter#3]
```
# ./ambientweatherexporter-linux-arm64 --port 9101 --station-name="Rain Hollow"
ambientweatherexporter version v1.5.2-0-g032bb36 built on Mon Feb 19 19:45:40 UTC 2024 with go version go1.22.0 linux/amd64
2024/03/03 11:38:51 Failed to parse incoming request: runtime error: index out of range [0] with length 0
```

It looks eerily familiar to the WS-2902 reported in #1, but I'm running the latest v1.5.2.

```
# nc -l 9101
GET /data/report/&PASSKEY=xxx&stationtype=AMBWeatherPro_V5.1.5&dateutc=2024-03-03+11:40:53&tempf=67.3&humidity=99&windspeedmph=0.00&windgustmph=0.00&maxdailygust=3.36&winddir=146&uv=0&solarradiation=0.02&hourlyrainin=0.071&eventrainin=0.051&dailyrainin=0.051&weeklyrainin=0.051&monthlyrainin=0.051&totalrainin=55.890&battout=1&tempinf=73.6&humidityin=62&baromrelin=30.171&baromabsin=30.056 HTTP/1.1
Host:192.168.1.5
Connection:close
User-Agent:AMBWeatherPro
```

**Comment by tedpearson** [comment]
It looks familiar because it's exactly the same issue. 🙄 I reintroduced the issue trying to sanitize log lines, which I did _before_ the error check instead of after it.

Fixed again in [v1.5.3](https://github.com/tedpearson/ambientweatherexporter/releases/tag/v1.5.3).

**Comment by iBotPeaches** [comment]
thanks! confirmed working.
